torch_np/_reductions.py

```diff
@@ -265,21 +374,19 @@ def average_weights(a, axis, w, keepdims=False):
     denominator = w.sum(axis, dtype=result_dtype)
     result = numerator / denominator
 
-    # keepdims
-    if keepdims:
-        result = _util.apply_keepdims(result, axis, a.ndim)
-
     return result, denominator
 
 
 def quantile(
```

Comment:
Not using `_deco_axis_expand` because it assumes the signature `a, axis` and here it's `a, q, axis` with an extra array_like `q`, correct? Might want to add a code comment.

Reply:
Yep. Will leave a comment.